src/workerd/jsg/modules-new.c++

```diff
@@ -378,41 +375,62 @@ class IsolateModuleRegistry final {
       auto module = entry.key.getHandle(js);
       auto status = module->GetStatus();
 
-      if (status == v8::Module::kEvaluated) {
-        return module->GetModuleNamespace().As<v8::Object>();
+      // If status is kErrored, that means a prior attempt to evaluate the module
```

Comment:
Q: Does the new module registry uses the compile cache that is generated from gen-compile-cache step?

Reply:
Not yet. The original implementation here predated the `gen-compile-cache` stuff and hasn't yet been updated for that. That will be one of the upcoming PRs to ensure that is all used.